internal/task/manager/getstatus.go

```diff
@@ -0,0 +1,39 @@
+package task
+
+import (
+	"context"
+
+	basegrpc "github.com/mikewurtz/taskman/internal/grpc"
+	basetask "github.com/mikewurtz/taskman/internal/task"
+)
+
+func (tm *TaskManager) GetTaskStatus(ctx context.Context, taskID string) (*Task, error) {
+	task, err := tm.GetTask(taskID)
+	if err != nil {
+		return nil, err
+	}
+
+	caller := ctx.Value(basegrpc.ClientIDKey).(string)
```

Comment:
Should we consider moving authorization to the gRPC level? Does it make sense to validate the owner of the job so far away from the gRPC server where these details are defined?

Reply:
I think moving it to the gRPC level is a good approach. It makes the library implementation cleaner and more focused on managing tasks like it is written to do.